bin/prover/src/prove/mod.rs

```diff
@@ -23,6 +44,10 @@ impl IntoResponse for ProveError {
     }
 }
 
-pub fn router() -> Router {
-    Router::new().route("/state-diff-commitment", post(state_diff_commitment::root))
-}
+pub fn router(app_state: &AppState) -> Router{
+    Router::new()
+        .route("/state-diff-commitment", post(state_diff_commitment::root))
+        .route("/state-diff-commitment/auth", get(crate::auth::generate_nonce))
```

Comment:
authetication is for the whole server, not specifically for `state-diff-commitment` program.
Endpoint for auth should be `/auth` instead of `/prove/state-diff-commitment/auth`